lib/debug.py

```diff
@@ -47,16 +47,74 @@ def init_log(self):
+        # Edition. Jetbrains has chosen to make remote debugging a Pro feature.
+        # To debug Inkscape python extensions, the extension code and Inkscape run
+        # independently of PyCharm, and only communicate with the debugger via a
+        # TCP socket. Thus, debugging is "remote," even if it's on the same machine,
+        # connected via the loopback interface.
+        #
+        # 1.     pip install pydev_pycharm
+        #
+        #    pydev_pycharm is versioned frequently. Jetbrains suggests installing
+        #    a version at least compatible with the current build. For example, if your
+        #    PyCharm build, as found in menu PyCharm -> About Pycharm is 223.8617.48,
+        #    you could do:
+        #        pip install pydevd-pycharm~=223.8617.48
+        #
+        # 2. From the Pycharm "Run" menu, choose "Edit Configurations..." and create a new
+        #    configuration. Set "IDE host name:" to  "localhost" and "Port:" to 5678.
+        #    You can leave the default settings for all other choices.
+        #
+        # 3. Touch a file named "DEBUG" at the top of your git repo, as above.
+        #
+        # 4. Create a symbolic link in the Inkscape extensions directory to the
+        #    top-level directory of your git repo. On a mac, for example:
+        #        cd ~/Library/Application\ Support/org.inkscape.Inkscape/config/inkscape/extensions/
+        #        ln -s <full path to the top level of your Ink/Stitch git repo>
+        #    On other architectures it may be:
+        #        cd ~/.config/inkscape/extensions
+        #        ln -s <full path to the top level of your Ink/Stitch git repo>
+        #    Remove any other Ink/Stitch files or references to Ink/Stitch from the
+        #    extensions directory, or you'll see duplicate entries in the Ink/Stitch
+        #    extensions menu in Inkscape.
+        #
+        # 5. In the execution env for Inkscape, set the environment variable
+        #    PYCHARM_REMOTE_DEBUG to any value, and launch Inkscape. If you're starting
+        #    Inkscape from the PyCharm Terminal pane, you can do:
+        #        export PYCHARM_REMOTE_DEBUG=true;inkscape
+        #
+        # 6. In Pycharm, either click on the green "bug" icon if visible in the upper
+        #    right or press Ctrl-D to start debugging.The PyCharm debugger pane will
+        #    display the message "Waiting for process connection..."
+        #
+        # 7. Do some action in Inkscape which invokes Ink/Stitch extension code, and the
+        #    debugger will be triggered. If you've left "Suspend after connect" checked
+        #    in the Run configuration, PyCharm will pause in the "self.log("Enabled
+        #    PyDev debugger.)" statement, below. Uncheck the box to have it continue
+        #    automatically to your first set breakpoint.
+
+        # Thanks to lexelby for the hints needed to puzzle this out.
```

Comment:
You're welcome, but let's punt this line, I'm already all over the source ;)

Reply:
:thumbsup: